datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -273,6 +280,7 @@ public void start() {
     // now that instance is started, make sure it doesn't miss any assignment created during
     // the slow startup
     _eventQueue.put(CoordinatorEvent.createHandleAssignmentChangeEvent());
+    _assignmentRetryCount.set(0);
```

Comment:
Extracting this into a method with variable, whether to reset counter, will ensure any future changes to not miss the reset.